iis-common/src/main/java/eu/dnetlib/iis/common/java/io/SequenceFileTextValueReader.java

```diff
@@ -0,0 +1,168 @@
+         * 
+         * @param fs hadoop filesystem
+         * @param seqPath sequence file path
+         * @throws IOException
+         */
+        public SequenceFileTextValueReader(final FileSystemPath path) 
+                        throws IOException {
+                this.fs = path.getFileSystem();
+                if (fs.isDirectory(path.getPath())) {
+                        fileIt = fs.listFiles(path.getPath(), false);
+                        sequenceReader = getNextSequenceReader();
+                } else {
+                        fileIt = null;
+                        sequenceReader = new Reader(fs.getConf(),
+                                        SequenceFile.Reader.file(path.getPath()));
+                }
+        }
+
+        final Reader getNextSequenceReader() throws IOException {
+                while (fileIt != null && fileIt.hasNext()) {
+                        LocatedFileStatus currentFileStatus = fileIt.next();
+                        if (isValidFile(currentFileStatus)) {
+                                return new Reader(this.fs.getConf(),
+                                                SequenceFile.Reader.file(currentFileStatus.getPath()));
+                        }
+                }
+//              fallback
+                return null;
+        }
+
+        /**
+         * Checks whether file is valid candidate.
+         * @param fileStatus
+         * @return true when valid, false otherwise
+         */
+        private final boolean isValidFile(LocatedFileStatus fileStatus) {
+                if (fileStatus.isFile()) {
+                        if (whitelistRegexp!=null) {
+                                return Pattern.matches(whitelistRegexp, 
+                                                fileStatus.getPath().getName());
+                        } else {
+                                return true;
+                        }
+                }
+//              fallback
+                return false;
+        }
+
+        /**
+         * Returns next data package.
+         * @return next data package
+         * @throws IOException
+         */
+        protected Text getNext() {
+                try {
+                        if (sequenceReader==null) {
+                                return null;
+                        }
+                        Writable key = (Writable) ReflectionUtils.newInstance(
+                                        sequenceReader.getKeyClass(), fs.getConf());
+                        Writable value = (Writable) ReflectionUtils.newInstance(
+                                        sequenceReader.getValueClass(), fs.getConf());
+                        if (sequenceReader.next(key, value)) {
+                        return (Text) value;
+                      } else {
+                          sequenceReader.close();
+                          sequenceReader = getNextSequenceReader();
+                          if (sequenceReader!=null) {
```

Comment:
`sequenceReader` is reassigned before second check what justifies its existence. It seems first `sequenceReader == null` check is currently required because `sequenceReader` is also instantiated by constructor and it might be `null`.